File: backend/app/schemas/hawk_eye.py
# PATH: backend/app/schemas/hawk_eye.py
# ───────────────────────────────
"""Cortex AI — HawkEye Schemas"""
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class HawkEyeSignal(BaseModel):
    name: str
    direction: Literal["buy", "sell", "neutral"]
    value: float
    timeframe: str


class HawkEyeResult(BaseModel):
    instrument_key: str
    trading_symbol: str
    exchange: str
    composite_signal: Literal["buy", "sell", "neutral"]
    composite_score: float
    timeframe_signals: list[HawkEyeSignal]
    last_price: float
    strength: Literal["strong", "moderate", "weak"]
    agreement_pct: float = Field(ge=0.0, le=100.0)
    scanned_at: str


class HawkEyeResponse(BaseModel):
    results: list[HawkEyeResult]
    total: int
    timeframes_scanned: list[str]
    scanned_at: str

File: frontend/src/types/hawk_eye.ts
/**
 * Cortex AI — HawkEye Radar Types
 * =================================
 * Typed contracts for the HawkEye multi-timeframe scanner.
 * Replaces `useState<any>(null)` and raw JSON.stringify display.
 */

export type SignalDirection = "buy" | "sell" | "neutral";
export type TimeframeStrength = "strong" | "moderate" | "weak";

export interface HawkEyeSignal {
  name: string;
  direction: SignalDirection;
  value: number;
  timeframe: string;
}

export interface HawkEyeResult {
  instrument_key: string;
  trading_symbol: string;
  exchange: string;
  composite_signal: SignalDirection;
  composite_score: number;
  timeframe_signals: HawkEyeSignal[];
  last_price: number;
  strength: TimeframeStrength;
  agreement_pct: number; // % of timeframes agreeing on direction
  scanned_at: string;
}

export interface HawkEyeResponse {
  results: HawkEyeResult[];
  total: number;
  timeframes_scanned: string[];
  scanned_at: string;
}